### ISSUE TITLE
Missing error message in case of improper command line

### ISSUE THREAD

**Issue by HolQue** [test-fullautomation/robotframework-testsuitesmanagement#96]
By mistake I provided a path to command line variable `variant`:

`--variable variant:"../config/exercise-08_config_variant1.json"`

Outcome:

In console there is no concrete error message (except the useless `Unknown exception occurs. Details: None`).

The log file contains a call stack only:

```
UNKNOWN - Unknown exception occurs. Details: None
DEBUG - Traceback (most recent call last):
  File "D:\RobotFramework\python39\lib\site-packages\RobotFramework_TestsuitesManagement\__init__.py", line 67, in run_keyword
    return DynamicCore.run_keyword(self, name, args, kwargs)
  File "D:\RobotFramework\python39\lib\site-packages\robotlibcore.py", line 105, in run_keyword
    return self.keywords[name](*args, **(kwargs or {}))
  File "D:\RobotFramework\python39\lib\site-packages\RobotFramework_TestsuitesManagement\Keywords\CSetup.py", line 81, in testsuite_setup
    BuiltIn().unknown(f"{CConfig.sLoadedCfgError}")
  File "D:\RobotFramework\python39\lib\site-packages\robotframework-4.1.3-py3.9.egg\robot\libraries\BuiltIn.py", line 562, in unknown
    raise UnknownAssertionError(msg) if msg else UnknownAssertionError()
```

But what is the problem? Expected is the name of a variant, that must be defined within the variant configuration file.
If there is no match, the error message should tell that a variant with this name is not defined.

**Comment by test-fullautomation** [comment]
A check need to happen if the variant is correct. If not, a list of possible variants need to be logged. Similar to what TML does.

**Comment by namsonx** [comment]
Hello Holger,

Thank you for your finding, I will improve it on next release.

Thank you,
Son

**Comment by namsonx** [comment]
Hello Holger,

I created pull-request #98 for this enhancement.

Thank you,
Son

**Comment by HolQue** [comment]
Hello Son,

with

`--variable variant:"./localconfig/exercise-pg_localconfig_bench1.json"`

the outcome now is:

```
[ ERROR ] Testsuite management - Loading configuration level 2 failed!
         The variant './localconfig/exercise-pg_localconfig_bench1.json' is not defined in 'C:\RobotTest\projects\ROBFW_AIO\RobotFramework_AIO\test\tutorial-test\playground\config\exercise-pg_variants.json'
[ ERROR ] Did not find configuration file: ''!
Test Case exercise-pg-2 :: exercise-pg-2                              | UNKNOWN |
Parent suite setup unknown:
Unknown exception occurs. Details: Testsuite management - Loading configuration level 2 failed!
         The variant './localconfig/exercise-pg_localconfig_bench1.json' is not defined in 'C:\RobotTest\projects\ROBFW_AIO\RobotFramework_AIO\test\tutorial-test\playground\config\exercise-pg_variants.json'
------------------------------------------------------------------------------
Exercise-Pg-2                                                         | UNKNOWN |
Suite setup unknown:
Unknown exception occurs. Details: Testsuite management - Loading configuration level 2 failed!
         The variant './localconfig/exercise-pg_localconfig_bench1.json' is not defined in 'C:\RobotTest\projects\ROBFW_AIO\RobotFramework_AIO\test\tutorial-test\playground\config\exercise-pg_variants.json'
```

(1)

It is obvious that` ./localconfig/exercise-pg_localconfig_bench1.json` cannot be the name of a variant. Therefore it looks a little bit curious to read that this variant is not defined. I suggest to follow also my hint in

https://github.com/test-fullautomation/robotframework-testsuitesmanagement/issues/95#issuecomment-1411604797

Variant names in command line (`variant`) should be checked against characters that are not allowed inside names (like path separators). This rule would detect errors like
`--variable variant:"D:/path/to/configfile.json"`
where accidently a path to a config file is provided instead of a variant name.

(2)

Regarding:
`[ ERROR ] Did not find configuration file: ''!`

Seems to be an aftereffect. If no concrete info is available you should remove this error message.

(3)

Please rearrange the order of output messages. The most relevant aspects should be listed at first, followed by aftereffects. Therefore you should not start with

`Loading configuration level 2 failed`

Here the most important aspect is that the variant name is invalid (or that a variant name is not defined). That a configuration cannot be loaded is an aftereffect and should be mentioned in one of the following lines of the error message.

(4)

See also
https://github.com/test-fullautomation/robotframework-testsuitesmanagement/issues/96#issuecomment-1408610862

"... a list of possible variants need to be logged"

(5)

Please remove the indentation of line `The variant ...` (the line is already long enough)

(6)

Like already mentioned earlier in other issues: The error message is listed three times. Why? This is too much. Error messages should be listed only once.

(7)

Please remove `Unknown exception occurs`. This is confusing. If an error handling is implemented to catch a certain type of error and if such an error occurs and the error handling provides a proper corresponding error message, then there is no need to label this as `Unknown exception`. The error is detected and therefore not unknown.

**Comment by namsonx** [comment]
Hello Holger,

I'm going to raise an error and terminate robot run in case there is an issue with configuration loading.
And the log will look like this:
.. code::
    ==============================================================================
    Exercise-06
    ==============================================================================
    [ ERROR ] Testsuite management - Loading configuration level 2 failed! 
             '/exercise-06/localconfig/exercise-06_localconfig_bench2.json' is invalid variant name!!!
             Please find the suitable variant in this list:
                - 'default'
                - 'variant1'
                - 'variant2'
    
    [ ERROR ] Execution stopped by user.

Does it look fine now? if yes, I will create a pull request :)

Thank you,
Son

**Comment by HolQue** [comment]
Hi Son,

much better now.

Please add '`an`' in `is invalid variant` ->` is an invalid variant`

Why `stopped by user`?

The computation is stopped because of an error, but it's not the user who explicitely stops something.

**Comment by namsonx** [comment]
Hello Holger,

Stopped by user because this robot run is force stopped by our testsuitesmanagement library.
I tried to use Process.terminate_all_processes(kill=True) but the log of testcases is shown as below:
.. code::
    ==============================================================================
    Exercise-06
    ==============================================================================
    [ ERROR ] Testsuite management - Loading configuration level 2 failed! 
             '/exercise-06/localconfig/exercise-06_localconfig_bench2.json' is invalid variant name!!!
             Please find the suitable variant in this list:
                - 'default'
                - 'variant1'
                - 'variant2'
    
    Test Case exercise-06 :: exercise-06                                  | UNKNOWN |
    Parent suite setup unknown:
    Unknown exception occurs. Details: Testsuite management - Loading configuration level 2 failed!
             '/exercise-06/localconfig/exercise-06_localconfig_bench2.json' is invalid variant name!!!
             Please find the suitable variant in this list:
                - 'default'
                - 'variant1'
                - 'variant2'
    ------------------------------------------------------------------------------
    Exercise-06                                                           | UNKNOWN |
    Suite setup unknown:
    Unknown exception occurs. Details: Testsuite management - Loading configuration level 2 failed!
             '/exercise-06/localconfig/exercise-06_localconfig_bench2.json' is invalid variant name!!!
             Please find the suitable variant in this list:
                - 'default'
                - 'variant1'
                - 'variant2'
    
    
    1 test, 0 passed, 0 failed, 1 unknown
    ==============================================================================
    Debug:   C:\Users\mas2hc\Desktop\data_storage\workspace\ROBFWAIO-OSS\debug_log.log
    Output:  C:\Users\mas2hc\Desktop\data_storage\workspace\ROBFWAIO-OSS\output.xml
    Log:     C:\Users\mas2hc\Desktop\data_storage\workspace\ROBFWAIO-OSS\log.html
    Report:  C:\Users\mas2hc\Desktop\data_storage\workspace\ROBFWAIO-OSS\report.html

**Comment by test-fullautomation** [comment]
As agreed in our conference for version 0.7.0 we only do BuildIn.unknown(logmessage) in this case.
logger.error is commented out.

Please do this for all exceptions the same way!

For 0.8.0 we create a new ticket to improve unknown exception message.

**Comment by namsonx** [comment]
Hello @test-fullautomation  and @HolQue ,

I created this pull-request #116  to update error message and unknow logs of handling testsuites management.
The log now look like below: I think it's much better while we find the solution to modify  Robot Framework core.
.. code::
    ==============================================================================
    Exercise-06
    ==============================================================================
    [ ERROR ] Testsuite management - Loading configuration level 2 failed! 
             The variant name '/exercise-06/localconfig/exercise-06_localconfig_bench2.json' is invalid.
             Please find the suitable variant in this list:
                - 'default'
                - 'variant1'
                - 'variant2'
    
    Test Case exercise-06 :: exercise-06                                  | UNKNOWN |
    Parent suite setup unknown:
    Unknown exception occurs. Details: Loading configuration level 2 failed!
    ------------------------------------------------------------------------------
    Exercise-06                                                           | UNKNOWN |
    Suite setup unknown:
    1 test, 0 passed, 0 failed, 1 unknown
    ==============================================================================
    Debug:   c:\Users\mas2hc\Desktop\data_storage\workspace\ROBFWAIO-OSS\debug_log.log
    Output:  c:\Users\mas2hc\Desktop\data_storage\workspace\ROBFWAIO-OSS\output.xml
    Log:     c:\Users\mas2hc\Desktop\data_storage\workspace\ROBFWAIO-OSS\log.html
    Report:  c:\Users\mas2hc\Desktop\data_storage\workspace\ROBFWAIO-OSS\report.html

+++++++++++++++++++++++
+++++++++++++++++++++++
or 
.. code::
    ==============================================================================
    Exercise-06
    ==============================================================================
    [ ERROR ] Parameter 'Maximum_version' with invalid value found in JSON configuration file!
    '1.0.0.3' does not match '^\\d+\\.\\d+\\.\\d+$'
    Test Case exercise-06 :: exercise-06                                  | UNKNOWN |
    Parent suite setup unknown:
    Unknown exception occurs. Details: JSON schema validation failed!
    ------------------------------------------------------------------------------
    Exercise-06                                                           | UNKNOWN |
    Suite setup unknown:
    Unknown exception occurs. Details: JSON schema validation failed!
    
    1 test, 0 passed, 0 failed, 1 unknown
    ==============================================================================
    Debug:   c:\Users\mas2hc\Desktop\data_storage\workspace\ROBFWAIO-OSS\debug_log.log
    Output:  c:\Users\mas2hc\Desktop\data_storage\workspace\ROBFWAIO-OSS\output.xml
    Log:     c:\Users\mas2hc\Desktop\data_storage\workspace\ROBFWAIO-OSS\log.html
    Report:  c:\Users\mas2hc\Desktop\data_storage\workspace\ROBFWAIO-OSS\report.html

+++++++++++++++++++++++
+++++++++++++++++++++++
or 
.. code::
    ==============================================================================
    Exercise-06
    ==============================================================================
    [ ERROR ] Verification against JSON schema failed: 'Additional properties are not allowed ('Testing' was unexpected)'
    [ ERROR ] Additional properties are not allowed!
                         Please put the additional params into 'params': { 'global': {...}
    Test Case exercise-06 :: exercise-06                                  | UNKNOWN |
    Parent suite setup unknown:
    Unknown exception occurs. Details: JSON schema validation failed!
    ------------------------------------------------------------------------------
    Exercise-06                                                           | UNKNOWN |
    Suite setup unknown:
    Unknown exception occurs. Details: JSON schema validation failed!
    
    1 test, 0 passed, 0 failed, 1 unknown
    ==============================================================================
    Debug:   c:\Users\mas2hc\Desktop\data_storage\workspace\ROBFWAIO-OSS\debug_log.log
    Output:  c:\Users\mas2hc\Desktop\data_storage\workspace\ROBFWAIO-OSS\output.xml
    Log:     c:\Users\mas2hc\Desktop\data_storage\workspace\ROBFWAIO-OSS\log.html
    Report:  c:\Users\mas2hc\Desktop\data_storage\workspace\ROBFWAIO-OSS\report.html

**Comment by namsonx** [comment]
Hello Thomas,

I created the pull-request #116  for this ticket.

Thank you,
Son

**Comment by test-fullautomation** [comment]
solved and already merged with #116
